src/pages/index.js

```diff
@@ -33,10 +46,56 @@ constants.profileEditButton.addEventListener("click", () => {
   profileFormValidator.resetValidation();
 });
 
-function handleProfileFormSubmit({ name, description }) {
-  user.setUserInfo({ name, description });
+function handleProfileFormSubmit({ name, about }) {
+  profileEditModal.setLoading(true);
+  api
+    .updateUserProfile({ name, about })
+    .then(() => {
+      user.setUserInfo({ name, about });
+      profileEditModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => profileEditModal.setLoading(false));
 }
 
+//User avatar
+const avatarEditModal = new ModalWithForm(
+  "#avatar-change-modal",
+  handleAvatarSubmit
+);
+avatarEditModal.setEventListeners();
+constants.userAvatarButton.addEventListener("click", () => {
+  avatarEditModal.open();
+});
+
+function handleAvatarSubmit({ link }) {
+  avatarEditModal.setLoading(true);
+  api
+    .updateAvatar({ link })
+    .then(() => {
+      user.setAvatar(link);
+    })
+    .then(() => {
```

Comment:
Needs correction: Here you should use `finally` as the loading should be ended in any case